### DISCUSSION TITLE
gltf格式模型加载显示问题

### DISCUSSION THREAD

**Discussion by luoyue123ddddddddddd** [galacean/discussion#118] [Engine] [answered]
在淘宝小程序里面加载gltf格式模型，在苹果手机上的表现：材质PBR里面的法线等贴图丢失，代码添加贴图也没用，大佬们有什么解决方法么？（安卓是正常的）

[watch0223.zip](https://github.com/galacean/discussion/files/14398889/watch0223.zip)
![图层 1](https://github.com/galacean/discussion/assets/109709383/ef2a7944-9668-4235-90b0-bb49ca4ba5d6)

**Comment by cptbtptpbcptdtptp** [comment]
麻烦补充一下引擎版本，淘宝小程序版本，以及是否在所有的 IOS 手机都出现？

**Reply by luoyue123ddddddddddd** [reply]
引擎0.9.22，淘宝小程序3.0.3，iPhone  6 、8、12、14，目前只测了这些机型

**Reply by zhuxudong** [reply, marked as answer]
这个文件图片的base64太长了都6M了，浏览器或者小程序对URL的长度通常有限制，但是这个限制的具体数值可能会因不同的操作系统或者小程序封装而异。一般来说，URL的长度限制在2,048个字符左右是比较常见的。超过这个长度的URL可能会被截断或导致加载错误。因此解决方法是将glTF文件的图片从 base64 替换成正常的 url，重新生成新的 glTF 文件，保证各平台兼容性。
![image](https://github.com/galacean/discussion/assets/17639043/31fbe785-3914-4baf-b818-7d059993bbc7)

![image](https://github.com/galacean/discussion/assets/17639043/0fc5162a-9f48-4714-8492-faeb39553eff)

**Reply by luoyue123ddddddddddd** [reply]
这些图片是模型材质里面的贴图图片？
